### PR TITLE
Fix package reference to system dacpac

### PR DESCRIPTION
**Pull request by zijchen**
Package references expect the dacpac name to be the same as package name, but is not the case for system dacpac packages. This change adds condition to check for `Microsoft.SqlServer.Dacpacs` package references and redirect to the master.dacpac within. Also adds `DacpacName` as an overridable atttribute to the PackageReference.

### CODE REVIEW

File: src/Microsoft.Build.Sql/sdk/Sdk.targets [hunk 1]
```diff
@@ -55,7 +55,17 @@
         <!-- https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Core/NuGet.Commands/RestoreCommand/Utility/BuildAssetsUtils.cs#L763 -->
         <PackageId>@(PackageReference->'%(Identity)'->Replace('.', '_'))</PackageId>
         <PackagePath>$(Pkg%(_ResolvedDacpacPackageReference.PackageId))</PackagePath>
-        <HintPath>%(_ResolvedDacpacPackageReference.PackagePath)/tools/%(Identity).dacpac</HintPath>
+        <IsSystemDacpac>@(PackageReference->'%(Identity)'->ToLowerInvariant()->Contains('microsoft.sqlserver.dacpacs'))</IsSystemDacpac>
+
+        <!-- File name of the dacpac inside the package -->
+        <DacpacName>%(PackageReference.DacpacName)</DacpacName>
+        <DacpacName Condition="'%(_ResolvedDacpacPackageReference.DacpacName)' == ''">%(PackageReference.Identity)</DacpacName>
+        <DacpacName Condition="'%(_ResolvedDacpacPackageReference.IsSystemDacpac)' == 'true'">master</DacpacName>
```
Comment (kisantia) [inline src/Microsoft.Build.Sql/sdk/Sdk.targets:63]: does `DacpacName` get set to msdb when the user is trying to reference msdb?

Comment (zijchen) [reply src/Microsoft.Build.Sql/sdk/Sdk.targets:63]: Yeah that's something users can set, we can test that during the bug bash

File: src/Microsoft.Build.Sql/sdk/Sdk.targets [hunk 2]
```diff
@@ -66,4 +76,4 @@
     </ItemGroup>
   </Target>
```
Comment (Benjin) [inline src/Microsoft.Build.Sql/sdk/Sdk.targets:78]: Extra line

File: src/Microsoft.Build.Sql/sdk/Sdk.targets (hunk 1 above)
[the remark this answers is not in the document]
Comment (zijchen) [reply src/Microsoft.Build.Sql/sdk/Sdk.targets]: Not by default, but users can override it

**Review by Benjin — APPROVED** [review]

**Pull request merged**